src/combine.cpp

```diff
@@ -352,6 +352,10 @@ void do_symbol_check(chunk_t *prev, chunk_t *pc, chunk_t *next)
    //         pc->text(), get_token_name(pc->type),
    //         next->text(), get_token_name(next->type));
 
+   if (next == NULL)
+   {
+      return;
```

Comment:
This will basically skip checking the last symbol. I think we need to check for null only where it's required.